projects/packages/forms/src/dashboard/components/InboxStatusToggle/index.js

```diff
@@ -0,0 +1,82 @@
+import {
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+import { useEntityRecords } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import './style.scss';
+
+/**
+ * Returns a formatted tab label with count.
+ *
+ * @param {string} label - The label for the tab.
+ * @param {number} count - The count to display.
+ * @return {string} The formatted label.
+ */
+function getTabLabel( label, count ) {
+	/* translators: 1: Tab label, 2: Count */
+	return sprintf( __( '%1$s (%2$s)', 'jetpack-forms' ), label, count || 0 );
+}
+
+/**
+ * Renders the status toggle for the inbox view.
+ *
+ * @param {object} props              - The component props.
+ * @param {object} props.currentQuery - The current query args from the parent.
+ * @return {JSX.Element} The status toggle component.
+ */
+export default function InboxStatusToggle( { currentQuery } ) {
+	const [ searchParams, setSearchParams ] = useSearchParams();
+	const status = searchParams.get( 'status' ) || 'inbox';
+	const queryBase = { search: '', page: 1, ...currentQuery, per_page: 1, _fields: 'id' };
+
+	const { totalItems: totalItemsInbox } = useEntityRecords( 'postType', 'feedback', {
+		...queryBase,
+		status: 'publish,draft',
+	} );
+	const { totalItems: totalItemsSpam } = useEntityRecords( 'postType', 'feedback', {
+		...queryBase,
+		status: 'spam',
+	} );
+	const { totalItems: totalItemsTrash } = useEntityRecords( 'postType', 'feedback', {
+		...queryBase,
+		status: 'trash',
+	} );
+
+	const statusTabs = [
+		{ label: getTabLabel( __( 'Inbox', 'jetpack-forms' ), totalItemsInbox ), value: 'inbox' },
+		{ label: getTabLabel( __( 'Spam', 'jetpack-forms' ), totalItemsSpam ), value: 'spam' },
+		{ label: getTabLabel( __( 'Trash', 'jetpack-forms' ), totalItemsTrash ), value: 'trash' },
+	];
+
+	const handleChange = useCallback(
+		newStatus => {
+			setSearchParams( prev => {
+				const params = new URLSearchParams( prev );
+				params.set( 'status', newStatus );
+				return params;
+			} );
+		},
+		[ setSearchParams ]
+	);
+
+	return (
+		<ToggleGroupControl
+			className="jp-forms__inbox-status-toggle"
+			value={ status }
+			onChange={ handleChange }
+		>
```

Comment:
FYI this is missing the `__next40pxDefaultSize` and `__nextHasNoMarginBottom` props shown in all our [code snippets](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-togglegroupcontrol--docs) for this component, meaning that this instance is likely logging deprecation warnings in the console right now. It is also missing an accessible label. We're planning on providing an Eslint plugin to help with stuff like this (ARC-158).

It's also missing the border around the component due to a missing `isBlock` prop which is the standard styling here, but I'm assuming that's partially intentional because you wanted the width to fit the content. This will be addressed in DS-194 so standard styling doesn't drift across usages.